phoenix-core/src/main/java/org/apache/phoenix/monitoring/connqueryservice/ConnectionQueryServicesMetricsManager.java

```diff
@@ -0,0 +1,375 @@
+        if (!isConnectionQueryServiceMetricsEnabled) {
+            return;
+        }
+        isConnectionQueryServiceMetricPublisherEnabled =
+                options.isConnectionQueryServiceMetricsPublisherEnabled();
+        LOGGER.info("Phoenix connection query service metrics publisher enabled status: "
+                + isConnectionQueryServiceMetricPublisherEnabled);
+    }
+
+    @VisibleForTesting
+    public static void setInstance(ConnectionQueryServicesMetricsManager metricsManager) {
+        connectionQueryServicesMetricsManager = metricsManager;
+    }
+
+    /**
+     * Method to provide instance of ConnectionQueryServiceMetricsManager(Create if needed in
+     * thread safe manner)
+     * @return returns instance of ConnectionQueryServicesMetricsManager for the said
+     *         ConnectionQueryService
+     */
+    public static ConnectionQueryServicesMetricsManager getInstance() {
+        ConnectionQueryServicesMetricsManager localRef = connectionQueryServicesMetricsManager;
+
+        if (localRef == null) {
+            synchronized (ConnectionQueryServicesMetricsManager.class) {
+                localRef = connectionQueryServicesMetricsManager;
+                if (localRef == null) {
+                    QueryServicesOptions options = QueryServicesOptions.withDefaults();
+                    localRef = connectionQueryServicesMetricsManager =
+                            new ConnectionQueryServicesMetricsManager(options);
+                    LOGGER.info("connection query service created object for metrics manager");
+                    if (isConnectionQueryServiceMetricsEnabled
+                            && isConnectionQueryServiceMetricPublisherEnabled) {
+                        String className = options.getConnectionProfileMetricsPublisherClass();
+                        if (className != null) {
+                            MetricServiceResolver mResolver = new MetricServiceResolver();
+                            LOGGER.info(String.format(
+                                    "connection query service metrics publisher className %s",
+                                    className));
+                            try {
+                                mPublisher = mResolver.instantiate(className);
+                                mPublisher.registerMetricProvider();
+                            } catch (Throwable e) {
+                                LOGGER.error("The exception from metric publish Function", e);
+                            }
+
+                        } else {
+                            LOGGER.warn("connection query service metrics publisher className"
+                                    + " can't be null");
+                        }
+                    }
+                }
+            }
+        }
+        return localRef;
+    }
+
+    private PhoenixConnectionQueryServiceMetrics getConnectionQueryServiceMetricsInstance(
```

Comment:
should this be the responsibility of PhoenixConnectionQueryServiceMetrics itself?